### DISCUSSION TITLE
Finding special Query Sets [Python] [Research]

### DISCUSSION THREAD

**Discussion by mutasDev** [github/codeql#9173] [Q&A] [answered]
Hey, in a research paper I stumbled on the usage of CodeQL with the testing suite "python-security-and-quality.qls" from offical CodeQL suites.

I'd like to re-use the insights from said paper, but I cannot find or download any query packs/suites with the given name.
The officiall "codeql/python-code-queries" query pack does not contain queries for all CWEs that are needed for my work.

Any insights or should I write the missing queries myself?

**Comment by atorralba** [comment, marked as answer]
That query suite can be found here: https://github.com/github/codeql/blob/main/python/ql/src/codeql-suites/python-security-and-quality.qls

**Reply by mutasDev** [reply]
Thanks for the reply, 

I though can't seem to download and run it on the CodeQL CLI.

Is `codeql pack download codeql-suites/python-security-and-quality` not correct?

**Reply by atorralba** [reply]
As far as I know there isn't a CodeQL query pack that provides that suite, but it can be used by cloning the CodeQL repository and referencing the `.qls` file in your filesystem when running the appropriate CLI command (like `codeql database analyze`).

**Reply by mutasDev** [reply]
That seemed to work, thank you very much !
